test/core/vm/state/state_tree_test.cpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "vm/state/state_tree.hpp"
+
+#include <gtest/gtest.h>
+#include "primitives/address/address_codec.hpp"
+#include "storage/hamt/hamt.hpp"
+#include "storage/ipfs/impl/in_memory_datastore.hpp"
+#include "testutil/cbor.hpp"
+#include "vm/actor/init_actor.hpp"
+
+using fc::primitives::BigInt;
+using fc::primitives::address::Address;
+using fc::storage::hamt::Hamt;
+using fc::storage::hamt::HamtError;
+using fc::vm::actor::Actor;
+using fc::vm::state::StateTree;
+
+auto kAddressId = Address::makeFromId(13);
+const Actor kActor{"010001020001"_cid, "010001020002"_cid, 3, BigInt(5)};
+
+class StateTreeTest : public ::testing::Test {
+ public:
+  std::shared_ptr<fc::storage::ipfs::IpfsDatastore> store_{
+      std::make_shared<fc::storage::ipfs::InMemoryDatastore>()};
+  StateTree tree_{store_};
+};
+
+/**
+ * @given State tree and actor state
+ * @when set
```

Comment:
pls make tests descriptions more detailed, 
e.g. 
@given State tree, actor state and predefined (key, value) pair
@when set key, value pair in the tree  
...
maybe my example is not good enough, just to show.